drivers/serial/uart_stm32.c

```diff
@@ -191,7 +191,7 @@ static inline void uart_stm32_set_baudrate(const struct device *dev, uint32_t ba
 		lpuartdiv = lpuartdiv_calc(clock_rate, baud_rate);
 		if (lpuartdiv < LPUART_BRR_MIN_VALUE || lpuartdiv > LPUART_BRR_MASK) {
 			LOG_ERR("Unable to set %s to %d", dev->name, baud_rate);
-			return;
+			return -EINVAL;
```

Comment:
Should the assertion below be replaced with error return code?
Ditto for lines 226/227.

Reply:
hmm. those first two assertions should never be tripped under normal operation. `lpuartdiv_calc` does the same calculation that happens in `LL_LPUART_SetBaudRate` (specifically the macro `__LL_LPUART_DIV`), and this range check is the same one that's asserted later. the only way i see that would happen is if writing to the registers fails.

the other assertion in the non-LPUART case (at L226) is one i overlooked. added additional checks for that case.

Reply:
Ok, thanks.